gdal/frmts/netcdf/netcdfsg.cpp

```diff
@@ -0,0 +1,1047 @@
+
+			// Read a single coord
+			int err = nc_get_var1_double(ncid, nodec_varIds[order], &real_ind, &data);
+
+			if(err != NC_NOERR)
+			{
+				throw new SG_Exception_BadPoint();
+			}
+
+			pt[order] = data;
+		}	
+
+		return *(this->pt_buffer);
+	}
+
+	size_t SGeometry::get_geometry_count()
+	{
+		if(type == POINT)
+		{
+			// If nodes global attribute is available, use that
+
+			// Otherwise, don't fail- use dimension length of one of x
+
+			if(this->nodec_varIds.size() < 1) return 0;
+
+			// If more than one dim, then error. Otherwise inquire its length and return that
+			int dims;
+			if(nc_inq_varndims(this->ncid, nodec_varIds[0], &dims) != NC_NOERR) return 0;
+			if(dims != 1) return 0;
+			
+			// Find which dimension is used for x
+			int index;
+			if(nc_inq_vardimid(this->ncid, nodec_varIds[0], &index) != NC_NOERR)
+			{
+				return 0;
+			}
+
+			// Finally find the length
+			size_t len;
+			if(nc_inq_dimlen(this->ncid, index, &len) != NC_NOERR)
+			{
+				return 0;
+			}
+			return len;	
+		}
+
+		else return this->node_counts.size();
+	}
+
+	/* serializeToWKB(SGeometry * sg)
+	 * Takes the geometry in SGeometry at a given index and converts it into WKB format.
+	 * Converting SGeometry into WKB automatically allocates the required buffer space
+	 * and returns a buffer that MUST be free'd
+	 */
+	void * SGeometry::serializeToWKB(size_t featureInd, int& wkbSize)
+	{		
+		void * ret = nullptr;
+
+		int nc = 0; int sb = 0;
+
+		// Points don't have node_count entry... only inspect and set node_counts if not a point
+		if(this->getGeometryType() != POINT)
+		{
+			nc = node_counts[featureInd];
+			sb = bound_list[featureInd];
+		}
+
+		// Serialization occurs differently depending on geometry
+		// The memory requirements also differ between geometries
+		switch(this->getGeometryType())
+		{
+			case POINT:
+				wkbSize = 1 + 4 + 16;
+				ret = new int8_t[wkbSize];
+				inPlaceSerialize_Point(this, featureInd, ret);
+				break;
+
+			case LINE:
+				wkbSize = 1 + 4 + 4 + 16 * nc;
+				ret = new uint8_t[wkbSize];
+				inPlaceSerialize_LineString(this, nc, sb, ret);
+				break;
+
+			case POLYGON:
+				// A polygon has:
+				// 1 byte header
+				// 4 byte Type (=3 for polygon)
+				// 4 byte ring count (1 (exterior) or 2 (exterior and interior)
+				// For each ring:
+				// 4 byte point count, 8 byte double x point count x 2 dimension
+				// (This is equivalent to requesting enough for all points and a point count header for each point)
+				// (Or 8 byte double x node count x 2 dimension + 4 byte point count x part_count)
+
+				// if interior ring, then assume that it will be a multipolygon (maybe future work?)
+				wkbSize = 1 + 4 + 4 + 4 + 16 * nc;
+				ret = new int8_t[wkbSize];
+				inPlaceSerialize_PolygonExtOnly(this, nc, sb, ret);
+				break;
+
+			case MULTIPOINT:
+				{
+					wkbSize = 1 + 4 + 4 + nc * (1 + 4 + 16);
+					ret = new int8_t[wkbSize];
+
+					void * worker = ret;
+					int8_t header = 1;
```

Comment:
1 is only valid on little-endian architecture. Same issue in other below occurences

Reply:
yeah I will make fixes according to your comment below, thanks! :+1: